diff --git a/CMakeLists.txt b/CMakeLists.txt
index 193e2a40c..7620fe490 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -192,7 +192,7 @@ if(BUILD_LAUNCHER)
 endif()
 message(STATUS "Fltk Libraries: ${FLTK_LIBRARIES}")
 
-set(JA2_INCLUDES "")
+set(JA2_INCLUDES "include/stracciatella")
 set(JA2_SOURCES "")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/src/externalized")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/src/game")
@@ -200,7 +200,6 @@ add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/src/sgp")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/dependencies/lib-lua")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/dependencies/lib-sol2")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/dependencies/lib-smacker")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/dependencies/lib-stracciatella")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/dependencies/lib-string_theory")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/dependencies/lib-miniaudio")
 if (MINGW)
@@ -265,7 +264,7 @@ set(
     ${SDL2_LIBRARY}
     ${GTEST_LIBRARIES}
     smacker
-    ${STRACCIATELLA_LIBRARIES}
+    lib/libstracciatella_c_api.a dl pthread
     string_theory-internal
     lua
 )
@@ -289,13 +288,11 @@ else()
 endif()
 target_link_libraries(${JA2_BINARY} ${JA2_LIBRARIES})
 
-add_dependencies(${JA2_BINARY} stracciatella)
 set_property(SOURCE ${CMAKE_SOURCE_DIR}/src/game/GameVersion.cc APPEND PROPERTY COMPILE_DEFINITIONS "GAME_VERSION=v${ja2-stracciatella_VERSION}")
 
 if(BUILD_LAUNCHER)
     add_executable(${LAUNCHER_BINARY} ${LAUNCHER_SOURCES})
-    target_link_libraries(${LAUNCHER_BINARY} ${FLTK_LIBRARIES} ${STRACCIATELLA_LIBRARIES} ${LAUNCHER_LIBRARIES})
-    add_dependencies(${LAUNCHER_BINARY} stracciatella)
+    target_link_libraries(${LAUNCHER_BINARY} ${FLTK_LIBRARIES} lib/libstracciatella_c_api.a dl pthread ${LAUNCHER_LIBRARIES})
 endif()
 
 macro(copy_assets_dir_to_ja2_binary_after_build DIR)
@@ -315,6 +312,8 @@ if(WITH_EDITOR_SLF)
     add_dependencies(${JA2_BINARY} download-editor-slf)
 endif()
 
+set(STRACCIATELLA_EXECUTABLES "bin/ja2-resource-pack")
+
 if (MSVC)
     string(REPLACE "SDL2main.lib" "SDL2.dll" SDL_DLL ${SDL2MAIN_LIBRARY})
     add_custom_command(TARGET ${JA2_BINARY} POST_BUILD
@@ -483,12 +482,6 @@ if (MINGW)
     install(SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/install-dlls-mingw.cmake")
 endif()
 
-if(APPLE)
-    file(GLOB APPLE_DIST_FILES "${CMAKE_CURRENT_SOURCE_DIR}/assets/distr-files-mac/*.txt")
-    install(FILES ${APPLE_DIST_FILES} DESTINATION .)
-    install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/lib-SDL2-2.0.8-macos/SDL2.framework DESTINATION .)
-endif()
-
 ## Build AppImage
 
 add_custom_target(package-appimage
